### PR TITLE
Replace AudioConnection#queue methods with AudioConnection#set methods

### LINKED ISSUE #469: Replace AudioConnection#queue(...) with AudioConnection#set(...)
Instead of having a primitive built-in queue, it might better to keep it simple and only allow users to set a single audio source. We can still provide a utility class that supports the user with the queue logic.

### PR DESCRIPTION
**Pull request by Mysterypotatoguy**
Resolves #469 

Unsure on the naming of these methods, perhaps they should just be `setAudioSource`, `removeAudioSource` etc..

### CODE REVIEW

File: javacord-core/src/main/java/org/javacord/core/audio/AudioConnectionImpl.java [hunk 1]
```diff
@@ -71,7 +71,7 @@
     /**
      * A queue with all audio sources for this connection.
      */
-    private final BlockingQueue<AudioSource> queue = new LinkedBlockingQueue<>();
+    private final BlockingQueue<AudioSource> queue = new LinkedBlockingQueue<>(1);
```
Comment (Bastian) [inline javacord-core/src/main/java/org/javacord/core/audio/AudioConnectionImpl.java:74]: Is there a reson to still have the queue?

Comment (Mysterypotatoguy) [reply javacord-core/src/main/java/org/javacord/core/audio/AudioConnectionImpl.java:74]: It's been some time, but I believe to maintain the ability to block while waiting for a source without resorting to wait/notify

Comment (Bastian) [reply javacord-core/src/main/java/org/javacord/core/audio/AudioConnectionImpl.java:74]: For reference
image
image

Tldr: Please create a helper class that abstracts this behaviour.

File: javacord-core/src/main/java/org/javacord/core/audio/AudioConnectionImpl.java [hunk 2]
```diff
@@ -307,38 +299,7 @@ public synchronized boolean tryConnect() {
...
-        }
-        return source;
-    }
-
-    /**
-     * Removes the current audio source.
-     */
-    public void removeCurrentSource() {
-        currentSource.set(null);
+        return currentSource.get();
```
Comment (Bastian) [inline javacord-core/src/main/java/org/javacord/core/audio/AudioConnectionImpl.java:302]:
```suggestion
        return currentSource.get(timeout, unit);
```
This completely ignores the `timeout` and `unit` parameter.

File: javacord-core/src/main/java/org/javacord/core/util/concurrent/BlockingReference.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * @param <V> The type of object referred to by this reference.
+ */
+public class BlockingReference<V> {
+
+    private final Lock lock = new ReentrantLock();
+
+    private final Condition hasValue = lock.newCondition();
+
+    private V value;
```
Comment (Bastian) [inline javacord-core/src/main/java/org/javacord/core/util/concurrent/BlockingReference.java:19]:
```suggestion
    private volatile V value;
```

File: javacord-core/src/main/java/org/javacord/core/audio/AudioConnectionImpl.java [hunk 4]
```diff
@@ -307,38 +299,7 @@ public synchronized boolean tryConnect() {
      * @throws InterruptedException If interrupted while waiting.
      */
     public AudioSource getCurrentAudioSourceBlocking(long timeout, TimeUnit unit) throws InterruptedException {
-        AudioSource source;
-        currentSourceLock.lock();
```
Comment (Bastian) [inline javacord-core/src/main/java/org/javacord/core/audio/AudioConnectionImpl.java:311]: The `currentSourceLock` field can be removed as it is no longer used anywhere.

File: javacord-core/src/main/java/org/javacord/core/audio/AudioConnectionImpl.java [hunk 5]
```diff
@@ -68,14 +59,9 @@
...
-     * The source that gets played at the moment.
-     */
-    private volatile AtomicReference<AudioSource> currentSource = new AtomicReference<>();
-
-    /**
-     * A queue with all audio sources for this connection.
+     * The source that is currently being played.
      */
-    private final BlockingQueue<AudioSource> queue = new LinkedBlockingQueue<>();
+    private volatile BlockingReference<AudioSource> currentSource = new BlockingReference<>();
```
Comment (Bastian) [inline javacord-core/src/main/java/org/javacord/core/audio/AudioConnectionImpl.java:64]:
```suggestion
    private final BlockingReference<AudioSource> currentSource = new BlockingReference<>();
```

**Comment by Bastian** [comment]
Thx :)
